crates/omnix-health/src/lib.rs

```diff
@@ -101,7 +101,27 @@ impl NixHealth {
         let code = res.report();
 
         if json_only {
-            let json: HashMap<_, _> = checks.iter().map(|(k, v)| (*k, v)).collect();
+            let nix_info = NixInfo::get()
+                .await
+                .as_ref()
+                .with_context(|| "Unable to gather nix info")?;
+
+            let sysinfo_json: serde_json::Value = vec![
+                ("system", nix_info.nix_config.system.value.to_string()),
+                ("os", nix_info.nix_env.os.to_string()),
+                ("nix-installer", nix_info.nix_env.installer.to_string()),
+                ("ram", nix_info.nix_env.total_memory.to_string()),
+                ("disk-space", nix_info.nix_env.total_disk_space.to_string()),
```

Comment:
At this point, I recommend encapsulating all this stuff into a new `struct`, and derive JSON directly from that without going through manual `Value` wiring up.

Using struct is less error-prone over time when we would be making changes.

Reply:
```rust
struct HealthOutput { 
  pub checks: ...,
  pub info: HealthEnvInfo
}

struct HealthEnvInfo {
  pub nix_installer: NixInstaller, 
}

/// https://serde.rs/enum-representations.html
#[serde(tag = "type")]
enum NixInstaller {
  DetSys(DetSysNixInstaller)
}
```

You may even move all of this to `json.rs` if necessary.

Reply:
Added `json.rs` in https://github.com/juspay/omnix/pull/404/commits/7c67f8a406de625fa1795ade2fece9fdde5f2fa7